aspnetcore/mvc/views/razor.md

```diff
@@ -442,43 +440,113 @@ public class _Views_Something_cshtml : RazorPage<dynamic>
 
-The `@using` directive adds the C# `using` directive to the generated view:
+```cshtml
+@attribute [Authorize]
+```
 
-[!code-cshtml[](razor/sample/Views/Home/Contact9.cshtml)]
+::: moniker range=">= aspnetcore-3.0"
 
-### @model
+### \@code
 
-The `@model` directive specifies the type of the model passed to a view:
+The `@code` directive enables a [Razor component](xref:blazor/components) to add C# members (fields, properties, and methods) to a component:
 
 ```cshtml
-@model TypeNameOfModel
+@code {
+    // C# members (fields, properties, and methods)
+}
 ```
 
-In an ASP.NET Core MVC app created with individual user accounts, the *Views/Account/Login.cshtml* view contains the following model declaration:
+For Razor components, `@code` is an alias of [@functions](#functions) and recommended over `@functions`. More than one `@code` directive is permissible.
+
+::: moniker-end
+
+### \@functions
+
+The `@functions` directive enables a Razor Page to add C# members (fields, properties, and methods) to a view:
 
 ```cshtml
-@model LoginViewModel
+@functions {
+    // C# members (fields, properties, and methods)
+}
 ```
 
-The class generated inherits from `RazorPage<dynamic>`:
+::: moniker range=">= aspnetcore-3.0"
 
-```csharp
-public class _Views_Account_Login_cshtml : RazorPage<LoginViewModel>
+> [!NOTE]
+> In [Razor components](xref:blazor/components), use `@code` over `@functions` to add C# members.
+
+::: moniker-end
+
+For example:
+
+[!code-cshtml[](razor/sample/Views/Home/Contact6.cshtml)]
+
+The code generates the following HTML markup:
+
+```html
+<div>From method: Hello</div>
 ```
 
-Razor exposes a `Model` property for accessing the model passed to the view:
+The following code is the generated Razor C# class:
+
+[!code-csharp[](razor/sample/Classes/Views_Home_Test_cshtml.cs?range=1-19)]
+
+::: moniker range=">= aspnetcore-3.0"
+
+`@functions` methods serve as templating methods when they have markup:
 
 ```cshtml
-<div>The Login Email: @Model.Email</div>
+@{
+    RenderName("Mahatma Gandhi");
+    RenderName("Martin Luther King, Jr.");
+}
+
+@functions {
+    private void RenderName(string name)
+    {
+        <p>Name: <strong>@name</strong></p>
+    }
+}
+```
+
+The code renders the following HTML:
+
+```html
+<p>Name: <strong>Mahatma Gandhi</strong></p>
+<p>Name: <strong>Martin Luther King, Jr.</strong></p>
 ```
 
-The `@model` directive specifies the type of this property. The directive specifies the `T` in `RazorPage<T>` that the generated class that the view derives from. If the `@model` directive isn't specified, the `Model` property is of type `dynamic`. The value of the model is passed from the controller to the view. For more information, see [Strongly typed models and the &commat;model keyword](xref:tutorials/first-mvc-app/adding-model#strongly-typed-models-and-the--keyword).
+::: moniker-end
+
+### \@implements
```

Comment:
This was added in 3.0, also it uses an `@code` in the body. I'd change it to `@functions` so it's .cshtml/.razor compatible.